### ISSUE TITLE
Migrate from CRA to Vite

### ISSUE THREAD

**Issue by Zenb0t** [kaiodeodato/pic_to_text#2]
Hey Kaio. Create React App has a lot of vulnerabilities, and it's painfully slow. We should think about migrating it to something like Vite.

Here is a [migrating tut](https://cathalmacdonnacha.com/migrating-from-create-react-app-cra-to-vite), and more about Vite [here](https://vitejs.dev/)

I can help you with this but should be fairly straightforward.

**Comment by GIT-Gizmo** [comment]
Hi, I'm happy to help with this, you can assign the issue to me if you don't mind

**Comment by kaiodeodato** [comment]
Hi man, cool! will be a honor! send me your linkedin for us to be in touch!

**Comment by GIT-Gizmo** [comment]
My pleasure.

Here is my linkedIn profile: https://linkedin.com/in/boluwatife-david

**Comment by GIT-Gizmo** [comment]
I have migrated to vite but the console return an error related to your supabase_url and supabase_key which isn't there you might have to put those in yourself. I also noticed you added your .env to gitignore which might also play a role in the error.

**Comment by Zenb0t** [comment]
Hey, sorry for jumping in, but this might be related to a few key differences between Vite and CRA. Importing env variables is different when using Vite. [Here]( https://vitejs.dev/guide/env-and-mode.html) is the docs regarding the difference. Have to insert VITE_ in front of the env vars that you want to expose. And import then as import.meta.env.variable_name

**Comment by GIT-Gizmo** [comment]
Yeah, I'm happy to get feedback and tips on how to solve the issue. I'll check the doc for solution, thanks 👍

**Comment by kaiodeodato** [comment]
Hello David, I think the problem is because the supabase database is gone, I had to delete because I was with the maximum of two project free already.
I need to come up with a solution that doesn't use Supabase to save the photo. do you have any idea?

**Comment by kaiodeodato** [comment]
I cleaned it up to make things clearer, main is in CRA and the backend is already corrected

**Comment by GIT-Gizmo** [comment]
Oh alright, sorry for not getting back to you soon enough, I was really busy today.

**Comment by kaiodeodato** [comment]
No problem, you will see that the repo restarted, but the code, now, is
corrected 💪

Em sex., 1 de set. de 2023 às 21:39, David Bolu ***@***.***>
escreveu:
